File: backzhaymer/models/task.js
const mongoose = require('mongoose');

const taskSchema = new mongoose.Schema({
    userId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true }, // Référence à l'utilisateur
    title: { type: String, required: true }, // Nouveau champ pour le titre
    description: { type: String, required: true },
    isCompleted: { type: Boolean, default: false },
    date: { type: String, required: true }, // Date pour la tâche
}, { timestamps: true }); // Ajoute les champs createdAt et updatedAt

module.exports = mongoose.model('Task', taskSchema);

File: ALZAHAIMER/app/src/main/java/com/ensa/testazahaimar/ui/model/Task.kt
package com.ensa.testazahaimar.ui.model

import java.io.Serializable

data class Task(
    val id: String,                // Identifiant unique de la tâche

    val title: String,             // Titre de la tâche
    val description: String,       // Description de la tâche
    var isCompleted: Boolean = false, // Statut de la tâche (complétée ou non)
    val date: String                 // Date de la tâche
)
